core/internal/registry.go

```diff
@@ -29,7 +30,7 @@ type ModuleInitializer struct {
 func ModulesByProtoMessageName() (map[protoreflect.FullName]*ModuleInitializer, error) {
 	res := map[protoreflect.FullName]*ModuleInitializer{}
 
-	for _, initializer := range ModuleRegistry {
+	for _, initializer := range maps.Values(ModuleRegistry) {
 		descriptor := initializer.ConfigProtoMessage.ProtoReflect().Descriptor()
 		fullName := descriptor.FullName()
 		if _, ok := res[fullName]; ok {
```

Comment:
This code isn't sorting the values, so AFAICT doesn't produce behavior any different than ranging over the map directly?